### PR TITLE
Rewrite CoC, incorporating the guidelines.

### PR DESCRIPTION
**Pull request by smalyshev**
This change introduces new CoC text, along the lines that I outlines in the mailing list. I have also incorporated the bulk of "Constructive Collaboration Guidelines", making them more general and not specific to RFCs. We may also have separate page on wiki with exact text of "Constructive Collaboration Guidelines" as pertains to RFCs, but I think for CoC it needs to be more generic.

Except for very light editing, I did not touch much beyond CoC, that has to be dealt with separately.

Fixes of bad English spelling, grammar and so on are extremely welcome.

### CODE REVIEW

File: RFC.rst [hunk 1]
```diff
@@ -14,92 +14,82 @@ The goal of this code of conduct is to give a clear signal what is expected of c
 
 This proposal consists of several parts:
 
-  * A general "Code of Conduct", which provides a general overview on what constitutes as unacceptable behaviour
-  * A set of guidelines to foster collaboration on proposed language features through RFCs, in addition to the formal RFC process that we already have at https://wiki.php.net/rfc/voting
-  * A set of procedures to follow if individuals wish to make complaints if other project members do not follow either the Code of Conduct or the Constructive Collaboration Guidelines.
-  * A Contributor Mediation Team to assess and act upon the complaints as received through the above procedures
-  * Possibilities for strong action if things really go awry regarding a project member. For example, if project members continue to persistently ignore the Collaboration Guidelines, or violate the Code of Conduct, after several rounds of mediation and/or complaints.
+  * A general "Code of Conduct", which provides a general overview on what is expected and unacceptable behaviour.
```
Comment (Jean85) [inline RFC.rst:17]: I would rephrase this as:
"on what is expected from project members and contributors and what constitutes unacceptable behaviour"
I think it's more readable, without changing the meaning.

**Comment by derickr** [comment]
Thanks for this - Now I've split up the documents, and imported Davey's changed CoC, this doesn't apply any more. Would you be so kind to go over the new set - and suggest changes (per subject) as PRs if the new text doesn't address your concerns? Sorry -- too many different replacements of the text confuse me.
